src/lib/tree/binary/BinaryMerkleTree.sol

```diff
@@ -77,6 +78,122 @@ library BinaryMerkleTree {
         return (computedHash == root, ErrorCodes.NoError);
     }
 
+    function verifyMulti(bytes32 root, BinaryMerkleMultiproof memory proof, /* maybe calldata */ bytes[] memory data)
```

Comment:
why calldata? I think memory is better as we will never need this data, no?

Reply:
I think there's zero downside to using `calldata` since it'll be copied to memory as needed. While if you declared it as `memory` it'll always copy to memory.

Reply:
just checked, apparently, using calldata is even cheaper than using memory in gas

Reply:
I am new to EVM development, so not totally sure.

I kept it as memory for consistency, because you already had it as memory in the NamespaceMerkleTree version. This is a helper function, so presumably, any external function that calls it will decide whether or not to make it calldata.

Reply:
If it's declared as `memory` then a copy to memory _must_ happen. If it's `calldata` then that's optional IIRC. So if it can be declared as `calldata` then it should be is my understanding.